PWGHF/vertexingHF/AliAnalysisTaskSEXic0Semileptonic.cxx

```diff
@@ -1896,28 +1841,47 @@ void AliAnalysisTaskSEXic0Semileptonic::DefineEventTree()
 	return;
 }//DefineEventTree
 
-//-------------------------------------------------------
-unsigned int AliAnalysisTaskSEXic0Semileptonic::GetEvID()
-{
-	/*
-	TString* currentfilename = (TObject*)(
-			(AliAnalysisManager::GetAnalysisManager()->GetInputEventHandler()->GetTree()->GetCurrentFile())
-			)->GetName();
-
-	if (!fFileName.EqualTo(currentfilename))
-	{
-		TString* fFileName = (TString*) currentfilename;
-		TObjArray *path = fFileName->Tokenize("/");
-		TString s = ((TObjString*)path->At( ((path->GetLast())-1) ))->GetString();
-		unsigned int fDirNumber = (unsigned int)s.Atoi();
-		delete path;
+void AliAnalysisTaskSEXic0Semileptonic::CheckOrigin(AliMCEvent* mcEvent, AliAODMCParticle *mcPart, Bool_t &c_flag, Bool_t &b_flag){
+	//from AliVertexingHFUtils
+  /// checking whether the mother of the particles come from a charm or a bottom quark
+
+  Int_t pdgGranma = 0;
+  Int_t mother = 0;
+  mother = mcPart->GetMother();
+  Int_t istep = 0;
+  Int_t abspdgGranma =0;
+  Bool_t isFromB=kFALSE;
+  Bool_t isQuarkFound=kFALSE;
+	c_flag=kFALSE;
+	b_flag=kFALSE;
+
+	while (mother >=0 ){
+    istep++;
+    AliAODMCParticle* mcGranma = (AliAODMCParticle*)mcEvent->GetTrack(mother);
+    if(mcGranma){
+			pdgGranma = mcGranma->GetPdgCode();
+	 		abspdgGranma = TMath::Abs(pdgGranma);
+	 		if ((abspdgGranma > 500 && abspdgGranma < 600) || (abspdgGranma > 5000 && abspdgGranma < 6000)) isFromB=kTRUE;
+	    if(abspdgGranma==4 || abspdgGranma==5) isQuarkFound=kTRUE;
+			mother = mcGranma->GetMother();
+    	}else{
+	      printf("CheckOrigin: Failed casting the mother particle!");
+	      break;
+    }
+  }
+
+	if(!isQuarkFound){
```

Comment:
I would suggest you not to consider particles that do not have a charm or beauty quark in the decay chain as we do for D and Lc, also because if you will run this task in p-Pb or Pb-Pb in the future, you will include HIJING signals. Hence I would suggest you either to reject those signal or add a flag to choose if keep them or not (analogous of `searchUpToQuark` in https://github.com/alisw/AliPhysics/blob/16e39139f8a8b04b2b3b2060f330e17f63502ab4/PWGHF/vertexingHF/AliVertexingHFUtils.h#L166).

Reply:
Hi @fgrosa , Now I understanding your motion. I modified it and will open PR. Thanks a lot!